git/filter_protocol.go

```diff
@@ -0,0 +1,114 @@
+package git
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strconv"
+	"strings"
+)
+
+type protocol struct {
```

Comment:
This can probably be renamed in a later PR (or maybe it already has, I haven't gotten that far yet!): `FilterProcessProtocol`?